### ISSUE TITLE
`pip install ddsp` slow on Colab (transitive dependencies again not pip-compileable)

### ISSUE THREAD

**Issue by bluenote10** [magenta/ddsp#460]
Just for your information, I've noticed that `pip install ddsp` is again taking ages on Colab. Also, trying to pip-compile its transitive dependencies currently fails with

```
$ pip-compile requirements.in
Could not find a version that matches protobuf==3.20.0,>=3.12.2,>=3.9.2,>=4.21.2 (from note-seq==0.0.5->ddsp==3.2.1->-r requirements.in (line 1))
Tried: 2.0.3, 2.3.0, 2.4.1, 2.5.0, 2.6.0, 2.6.1, 3.0.0, 3.0.0, 3.0.0, 3.1.0, 3.1.0.post1, 3.1.0.post1, 3.2.0, 3.2.0, 3.2.0, 3.3.0, 3.3.0, 3.4.0, 3.4.0, 3.4.0, 3.5.0.post1, 3.5.0.post1, 3.5.0.post1, 3.5.1, 3.5.1, 3.5.1, 3.5.2, 3.5.2, 3.5.2, 3.5.2.post1, 3.5.2.post1, 3.5.2.post1, 3.6.0, 3.6.0, 3.6.0, 3.6.1, 3.6.1, 3.6.1, 3.7.0, 3.7.0, 3.7.0, 3.7.1, 3.7.1, 3.7.1, 3.8.0, 3.8.0, 3.8.0, 3.9.0, 3.9.0, 3.9.0, 3.9.1, 3.9.1, 3.9.1, 3.9.2, 3.9.2, 3.9.2, 3.10.0, 3.10.0, 3.10.0, 3.11.0, 3.11.0, 3.11.0, 3.11.1, 3.11.1, 3.11.1, 3.11.2, 3.11.2, 3.11.2, 3.11.3, 3.11.3, 3.11.3, 3.12.0, 3.12.1, 3.12.2, 3.12.2, 3.12.2, 3.12.4, 3.12.4, 3.12.4, 3.13.0, 3.13.0, 3.13.0, 3.14.0, 3.14.0, 3.14.0, 3.15.0, 3.15.0, 3.15.0, 3.15.1, 3.15.1, 3.15.1, 3.15.2, 3.15.2, 3.15.2, 3.15.3, 3.15.3, 3.15.3, 3.15.4, 3.15.4, 3.15.4, 3.15.5, 3.15.5, 3.15.5, 3.15.6, 3.15.6, 3.15.6, 3.15.7, 3.15.7, 3.15.7, 3.15.8, 3.15.8, 3.15.8, 3.16.0, 3.16.0, 3.16.0, 3.17.0, 3.17.0, 3.17.0, 3.17.1, 3.17.1, 3.17.1, 3.17.2, 3.17.2, 3.17.2, 3.17.3, 3.17.3, 3.17.3, 3.18.0, 3.18.0, 3.18.0, 3.18.1, 3.18.1, 3.18.1, 3.19.0, 3.19.0, 3.19.0, 3.19.1, 3.19.1, 3.19.1, 3.19.2, 3.19.2, 3.19.2, 3.19.3, 3.19.3, 3.19.3, 3.19.4, 3.19.4, 3.19.4, 4.21.0, 4.21.0, 4.21.0
Skipped pre-versions: 2.0.0b0, 3.0.0a2, 3.0.0a3, 3.0.0b1, 3.0.0b1.post1, 3.0.0b1.post2, 3.0.0b2, 3.0.0b2, 3.0.0b2.post1, 3.0.0b2.post1, 3.0.0b2.post2, 3.0.0b2.post2, 3.0.0b3, 3.0.0b4, 3.0.0b4, 3.2.0rc1, 3.2.0rc1, 3.2.0rc1.post1, 3.2.0rc1.post1, 3.2.0rc2, 3.2.0rc2, 3.7.0rc2, 3.7.0rc2, 3.7.0rc2, 3.7.0rc3, 3.7.0rc3, 3.7.0rc3, 3.8.0rc1, 3.8.0rc1, 3.8.0rc1, 3.9.0rc1, 3.9.0rc1, 3.9.0rc1, 3.10.0rc1, 3.10.0rc1, 3.10.0rc1, 3.11.0rc1, 3.11.0rc1, 3.11.0rc1, 3.11.0rc2, 3.11.0rc2, 3.11.0rc2, 3.12.0rc1, 3.12.0rc2, 3.13.0rc3, 3.13.0rc3, 3.13.0rc3, 3.14.0rc1, 3.14.0rc1, 3.14.0rc1, 3.14.0rc2, 3.14.0rc2, 3.14.0rc2, 3.14.0rc3, 3.14.0rc3, 3.14.0rc3, 3.15.0rc1, 3.15.0rc1, 3.15.0rc1, 3.15.0rc2, 3.15.0rc2, 3.15.0rc2, 3.16.0rc1, 3.16.0rc1, 3.16.0rc1, 3.16.0rc2, 3.16.0rc2, 3.16.0rc2, 3.17.0rc1, 3.17.0rc1, 3.17.0rc1, 3.17.0rc2, 3.17.0rc2, 3.17.0rc2, 3.18.0rc1, 3.18.0rc1, 3.18.0rc1, 3.18.0rc2, 3.18.0rc2, 3.18.0rc2, 3.19.0rc1, 3.19.0rc1, 3.19.0rc1, 3.19.0rc2, 3.19.0rc2, 3.19.0rc2, 4.0.0rc1, 4.0.0rc1, 4.0.0rc1, 4.0.0rc2, 4.0.0rc2, 4.0.0rc2, 4.21.0rc1, 4.21.0rc1, 4.21.0rc1, 4.21.0rc2, 4.21.0rc2, 4.21.0rc2
There are incompatible versions in the resolved dependencies:
  protobuf (from google-cloud-storage==2.0.0->ddsp==3.2.1->-r requirements.in (line 1))
  protobuf>=3.12.2 (from tensorflow-datasets==4.5.2->ddsp==3.2.1->-r requirements.in (line 1))
  protobuf>=4.21.2 (from note-seq==0.0.5->ddsp==3.2.1->-r requirements.in (line 1))
  protobuf==3.20.0 (from tensorflowjs==3.19.0->ddsp==3.2.1->-r requirements.in (line 1))
  protobuf>=3.9.2 (from tensorflow==2.6.2->ddsp==3.2.1->-r requirements.in (line 1))
```

which hints to why the pip install is so slow.

Looking a bit further, I've noticed that tensorflowjs has basically dropped Python 3.6 support, which of course causes friction on Colab due to being Python 3.6 based. I've raised an upstream issue to discuss the problem there: https://github.com/tensorflow/tfjs/issues/6698

As a temporary work-around people could try
```
pip install ddsp "note-seq<0.0.4" "tensorflowjs<3.19"
```
which gives pip a few additional version constraints that avoid pip going into its backtracking mode (at least as of the time of writing).

**Comment by nglazyrin** [comment]
Thank you @bluenote10 ! Plain "pip install" on a clean conda environment could not complete overnight, while your workaround completes in a few minutes.

**Comment by jesseengel** [comment]
This should be fixed with #472. I just did a fresh conda install of ddsp and it was done in a minute or two.
